src/components/icon/icon-provider.ts

```diff
@@ -0,0 +1,255 @@
+import {Injectable, Renderer} from 'angular2/core';
+import {Http, Response, HTTP_PROVIDERS} from 'angular2/http';
+import {AsyncSubject, Observer, Observable} from 'rxjs/Rx';
+
+/**
+  *  Configuration for an icon, possibly including the cached SVG element.
+  */
+class IconConfig {
+  svgElement: SVGElement = null;
+  constructor(public url: string, public viewBoxSize: number) {
+  }
+}
+
+@Injectable()
+export class MdIconProvider {
```

Comment:
I'm trying to come up with a better name for this class, since it would be good to move away from the Angular 1 "provider" concept. What do you think about `MdIconRegistry` ?